app/src/main/java/com/github/codetanzania/ui/fragment/MapboxBaseFragment.java

```diff
@@ -0,0 +1,177 @@
+        mMapView.onStop();
+    }
+
+    @Override
+    public void onLowMemory() {
+        super.onLowMemory();
+        mMapView.onLowMemory();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        mMapView.onDestroy();
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        mMapView.onSaveInstanceState(outState);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mLocationTracker.respondToActivityResult(requestCode, resultCode);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults) {
+        mLocationTracker.respondToPermissions(requestCode, grantResults);
+    }
+
+    private void updateCamera() {
+        mMapView.setCameraDistance(10);
+        CameraPosition position;
+        if (mLocationFoundPreviously) {
+            position = new CameraPosition.Builder()
+                    .target(mLatLng) // Sets the new camera position
+                    .build(); // Creates a CameraPosition from the builder
+        }
+        else {
+            mLocationFoundPreviously = true;
+            position = new CameraPosition.Builder()
+                    .target(mLatLng) // Sets the new camera position
+                    .zoom(15) // Sets the zoom
+                    //.bearing(180) // Rotate the camera
+                    //.tilt(30) // Set the camera tilt
+                    .build(); // Creates a CameraPosition from the builder
+        }
+
+        mMapboxMap.animateCamera(CameraUpdateFactory
+                .newCameraPosition(position), 7000);
+    }
+
+    private void addMarker() {
+        mMarker = new MarkerOptions()
+                .position(mLatLng)
+                .title("Location")
+                .snippet("Welcome to you");
+        mMapboxMap.addMarker(mMarker);
+    }
+
+    private void updateMarker() {
+        mMapboxMap.clear();
+        mMarker = new MarkerOptions()
+                .position(mLatLng)
+                .title("Location")
+                .snippet("Welcome to you");
+        mMapboxMap.addMarker(mMarker);
+    }
+
+    @Override
+    public void onMapReady(MapboxMap mapboxMap) {
+        mMapboxMap = mapboxMap;
+
+        mLocationTracker = new LocationTracker(getActivity());
+        mLocationTracker.start(this);
+    }
+
+    @Override
+    public void onLocationChanged(Location location) {
+        mLatLng = new LatLng(location.getLatitude(), location.getLongitude());
+        updateCamera();
+        if (mMarker != null) {
+            updateMarker();
+        }else{
+            addMarker();
+        }
+    }
+
+    @Override
+    public void onPermissionDenied() {
+        Toast.makeText(getActivity(), "DAWASCO needs a location to find the problem. GPS is required. Enable location and try again.", Toast.LENGTH_LONG).show();
```

Comment:
Hard coded stringz should be moved to the string resource. This makes it easier to do i10L